signal-desktop.sh

```diff
@@ -60,6 +84,19 @@ if [[ "${SIGNAL_PASSWORD_STORE}" == "basic" ]]; then
     fi
 fi
 
+# Explain filesystem=host upon the first run
+EXPLAIN_FILESYSTEM_HOST_FILE="${XDG_CONFIG_HOME}/Signal/explanation-filesystem-access-shown"
+if [[ ! -f "${EXPLAIN_FILESYSTEM_HOST_FILE}" ]]; then
+    acceptAccess="$(explain_filesystem_access)"
+    touch "${EXPLAIN_FILESYSTEM_HOST_FILE}"
```

Comment:
I guess this will fail on first start, because the `Signal` directory in `$XDG_CONFIG_HOME` does not exist at this point. Also, maybe this file should only be written if the user accepted the dialog.

Reply:
Same as above.

Reply:
Thanks.

It is not possible to determine if we have full filesystem=host access during runtime inside Signal. If the user clicked _no_ to exit Signal and then changed the permissions, there is no way to know. The next time Signal starts, it would again ask _yes_ or _no_. 

_no_ means cancel but the semantics of _yes_ would change that it does not actually use filesystem=host but the user-defined permission. I'm reworking the logic: _no_ always means cancel and _yes_ means accept file access or user manually changed permissions.

Reply:
@bermeitinger-b, I'm sorry I didn't had the time to implement it myself, but what you did is exactly what I had in my mind. Thanks a lot for creating and finishing this PR so that newer signal releases can be published again.